client/src/utils/APIUtils.js

```diff
@@ -159,4 +159,20 @@ export default class APIUtils {
             throw {status: response.status, message: data.error};
         }
     }
+
+    static fetchUserInterests = async () => {
+        const response = await fetch('http://localhost:3000/user/interests', {
+            method: "GET",
```

Comment:
Maybe I missed something, but this implementation looks identical to fetchRootInterests.

I think it's worth adding a query parameter for depth instead of having to maintain 2 functions for retrieving interests.

Examples:
```
Return All
http://localhost:3000/user/interests -> returns all

Root nodes
http://localhost:3000/user/interests?depth=0 

Various depths
http://localhost:3000/user/interests?depth=1
http://localhost:3000/user/interests/:interestId?depth=1
```